### PR TITLE
Update gitignore

### PR DESCRIPTION
**Pull request by svettwer**
Hi!

I updated the gitignore file based on [gitignore.io](https://gitignore.io) for various tools and IDEs. This will allow the developers to share e.g IDE specific configuration via GitHub so that for example licence and copyright updates are handled consistently.

### CODE REVIEW

**Comment by svettwer** [comment]
It's just part of the projects configuration. This should speed up the indexing/scanning when checking out the project for the first time.

**Comment by christophd** [comment]
I can imagine that these files are subject to change from OS to OS and IntelliJ versions so you end up having lots of merge conflicts. Also when using the same IntelliJ project for multiple branches this gets really complicated

**Comment by christophd** [comment]
Every time I saw these IDE files being part of the version control it was a mess and we ended up in removing those files from version control

**Comment by svettwer** [comment]
I gave the [jetbrains suggestions](https://intellij-support.jetbrains.com/hc/en-us/articles/206544839-How-to-manage-projects-under-Version-Control-Systems) of handling the .idea folder in VCS a read again.

>you may consider not to share the following: 
> - .iml files and .idea/modules.xml file for the Gradle or Maven based projects, since these files will be generated on import
> [...]

So I'll ignore them again.

File: .idea/compiler.xml [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+        <sourceTestOutputDir name="target/generated-test-sources/test-annotations" />
+        <outputRelativeToContentRoot value="true" />
+        <module name="citrus-model-mail" />
+        <module name="citrus-core" />
+        <module name="citrus-websocket" />
+        <module name="citrus-model-jdbc" />
+      </profile>
+    </annotationProcessing>
+    <bytecodeTargetLevel>
+      <module name="citrus" target="1.5" />
```
Comment (christophd) [inline .idea/compiler.xml]: 1.5?

File: .idea/compiler.xml [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+      <module name="citrus-core" target="1.8" />
+      <module name="citrus-model" target="1.5" />
```
Comment (christophd) [inline .idea/compiler.xml]: 1.5?

File: .idea/copyright/profiles_settings.xml [hunk 3]
```diff
@@ -0,0 +1,3 @@
+<component name="CopyrightManager">
+  <settings default="Apache 2" />
```
Comment (christophd) [inline .idea/copyright/profiles_settings.xml]: Users might have another copyright manager name configured in IDEA

File: .idea/misc.xml [hunk 4]
```diff
@@ -0,0 +1,129 @@
...
+            <id>TestNGJava</id>
+          </State>
+          <State>
+            <id>Threading issuesJava</id>
+          </State>
+        </expanded-state>
+      </profile-state>
+    </entry>
+  </component>
+  <component name="ProjectRootManager" version="2" languageLevel="JDK_1_8" project-jdk-name="1.8" project-jdk-type="JavaSDK" />
```
Comment (christophd) [inline .idea/misc.xml]: Users might have another name for the 1.8 JDK configured

File: .idea/modules.xml [hunk 5]
```diff
@@ -0,0 +1,13 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<project version="4">
+  <component name="ProjectModuleManager">
+    <modules>
```
Comment (christophd) [inline .idea/modules.xml]: This list of modules looks incomplete to me

File: .idea/sonarlint-state.xml [hunk 6]
```diff
@@ -0,0 +1,6 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<project version="4">
+  <component name="SonarLintProjectState">
+    <lastEventPolling>1557393725856</lastEventPolling>
```
Comment (christophd) [inline .idea/sonarlint-state.xml]: This timestamp will continuously change for different users, so you will have lots of merge conflicts

**Comment by svettwer** [comment]
There are some benefits for new users checking out the repository especially when creating new files that require a licence notice etc. 
I think the discussion becomes a little excessive and subjective as some statements show.

> Not a fan of adding all these IDEA related files to the version control.
> ...
> I would not like to have all those files as part of the version control to be honest.

I would like to talk about objective pros and cons or just take a look at other projects and how they handle such files. I think this is something I could have done in before to avoid such tedious discussions.

- [cucumber-jvm](https://github.com/cucumber/cucumber-jvm): No .idea folder
- [gradle](https://github.com/gradle/gradle): Shares code styles and inspection profiles
- [maven](https://github.com/apache/maven): No .idea folder
- [graal](https://github.com/oracle/graal): No .idea folder
- [RxJava](https://github.com/ReactiveX/RxJava): No .idea folder
- [reactor core](https://github.com/reactor/reactor-core): No .idea folder

So even if I have currently no information about their motivation to not check in such .idea files, there is a tendency to just leave those files out and to go without the small benefits the check in provides.

**Comment by christophd** [comment]
I am sorry to hear that my comments gave the impression of being `excessive and subjective`. I just wanted to make clear that I see lots of problems coming up as I have experienced in many other projects doing this.

Actually it was not about the `.gitignore` you could have merged this it was all about the IDE specific files that should not go to version control in my point of view.

**Comment by christophd** [comment]
Would have enjoyed to get more opinions on this from the other team members and community

**Pull request closed without merging**